frontend/stmt_inferrer.py

```diff
@@ -321,12 +321,19 @@ def _infer_func_def(node, context, solver):
     """Infer the type for a function definition"""
```

Comment:
IMO it would be fine to leave it like this for the moment and look at generic functions later, otherwise we'll have to keep this PR around forever.